images/basic/hostoverlaytest/Dockerfile

```diff
@@ -3,6 +3,6 @@ FROM ubuntu:bionic
 WORKDIR /root
 COPY . .
 
-RUN apt-get update && apt-get install -y gcc
+RUN apt-get update && apt-get --no-install-recommends install -y apt-utils ca-certificates gcc
```

Comment:
Is apt-utils and ca-certificates really needed by the various packages? I don't think any of these images use any tools from apt-utils or make requests to https web servers etc.

Reply:
I added "ca-certificates" to avoid any issue in web requests using curl, wget, or even git clone because without this package they fail and added "apt-utils" to make package configuration fast after installation through apt. We can ignore these as leaving them here doesn't make any difference and also avoid the above mentioned issue.

Reply:
1) We may be actually relying on the recommends for gcc, which includes libc-dev. Has you tested that these images actually work?
2) Isn't this actually makign all the images *bigger*, in addition to making the Dockerfile slightly more confusing? It's likely making this image slightly bigger.

In fact, I've checked about half the cases below. The only extra packages that seem to be installed here are ssl-cert (with apache2), ca-certificates, patch & less (with git). Conceivably, dropping patch may also suffer from (1).

I'm all clean ups, but have you tested that this is really an improvement?